### PR TITLE
Permalink must correctly process works with fulfillments

### PR DESCRIPTION
**Pull request by vbessonov**
## Description

This PR fixes a regression introduced in [SIMPLY-3354](https://jira.nypl.org/browse/SIMPLY-3354) when `permalink` route fails for works with fulfillments.

## Motivation and Context

[SIMPLY-3419](https://jira.nypl.org/browse/SIMPLY-3419)

## Checklist:

- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by leonardr — COMMENTED** [review]
We definitely shouldn't be treating a LicensePoolDeliveryMechanism as though it were a FulfillmentInfo. But I'm not convinced this is the right solution. I suspect what we want inside `permalink` is to always pass in `fulfillment=None`.

I say this because "converting a LicensePoolDeliveryMechanism into a FulfillmentInfo" can be an extremely complicated vendor-specific task that might result in multiple calls to API endpoints. You generally only want to do that when you are calling `CirculationAPI.fulfill`, because `fulfill` handles all the edge cases and lets vendor-specific code run.

I think the `fulfillment` in `single_item_feed` is designed for a situation where _you already called CirculationAPI.fulfill()_ and the thing you got back can't be sent directly to the user -- it needs to be turned into an OPDS entry first.

`single_item_feed` is called in a couple places apart from `permalink` (which calls `single_item_feed` as of your recent PR).

1. When someone asks to fulfill a book in a streaming HTML format (within CirculationAPI.fulfill itself). We need to make a separate request to the API to find the time-limited URL of the web page that we can give to the patron. Then we generate an <entry> tag that includes that link. This is the only situation I could find where `fulfill` was not None, and it happened _inside_ a call to `fulfill`.
2. In LoanController.revoke and LoanController.detail. In this case `fulfill` is None. (LoanController.detail actually seems very similar to the changes you made to permalink, but I'm not sure where it's used.)
3. In SharedCollectionLoanController.loan_info, hold_info, and borrow. In these case `fulfill` is always None.

Basically, since you are not actually fulfilling a loan here, only giving information about a loan, it should not be necessary to provide `fulfill` at all.

**Comment by vbessonov** [comment]
Thank you for noticing this. I just realised that sending `fulfillment` to `single_item_feed` in `permalink` could lead to really bad consequences. For example, in the case of self-hosted content protected by presigned S3 URLs it would lead to having unsigned URLs in an OPDS catalog entry.

File: tests/test_controller.py [hunk 1]
```diff
@@ -3109,6 +3061,96 @@ def test_permalink_returns_fulfillment_links_for_authenticated_patrons_with_loan
...
+                pool.identifier.identifier,
+                datetime.datetime.utcnow(),
+                datetime.datetime.utcnow() + datetime.timedelta(seconds=3600),
+            )
+            self.manager.d_circulation.queue_checkout(
+                pool,
+                loan_info
+            )
+
+            fulfillment = FulfillmentInfo(
```
Comment (leonardr) [inline tests/test_controller.py:3105]: It should no longer be necessary to create a FulfillmentInfo or a Resource as part of test setup, since it's no longer considered when generating the OPDS.

**Review by leonardr — APPROVED** [review]
I think there's some test setup that can be removed but otherwise this looks good.

File: tests/test_controller.py (hunk 1 above)
Comment (vbessonov) [reply tests/test_controller.py:3105]: The following code requires a `FulfillmentInfo` object:
```python
self.manager.loans.fulfill(
    pool.id, delivery_mechanism.delivery_mechanism.id,
)
```
It might make sense to remove the whole test case but I would like to still keep it to make sure that the code correctly processes works with fulfillments (by just ignoring them).
